### ISSUE TITLE
Running on Office31

### ISSUE THREAD

**Issue by RayWangWR** [erictzeng/adda#11]
Thanks for your source code.

I am trying to run it with the office dataset according to the setting in the paper using AlexNet, but the result is only getting worse during training. Could you please also release the code for office dataset? 

Thank you very much.

**Comment by RayWangWR** [comment]
I am adaption from amazon to webcam. During training, the validation accuracy on webcam first increased a little bit, then it began to drop very fast after about 20 iterations.

**Comment by LamForest** [comment]
One possible solution is to use a small lr for target encoder(e.g. 1e-5) and reduce training epoch. 
My setting:
ResNet-50 for source and target encoder.
lr for source encoder : 1e-3 this doesnot matter
lr for discriminator: 1e-3
lr for target encoder: 1e-5
trainin epoch : 6

**Comment by LamForest** [comment]
BTW, I use https://github.com/corenel/pytorch-adda not this offical Tensorflow. My result (A->W) is 81% using above setting.

**Comment by RayWangWR** [comment]
Hi Tianlin-Gao,

Thanks for your reply. May I know which layer are you matching and do you fix some layers in the source/target encoder to avoid overfitting? Also, can you tell me the number/size of the hidden layer of your discriminator and which optimizer are you using, parameters for the optimizer? Or would you mind sharing your code for adda on Office?

I am sorry that this has been killing me for days. I just want to get adda work and build my model based on it. I would be really thankful if you can provide more information on that implementation. 

Thanks,
Rui
rw161@duke.edu

**Comment by RayWangWR** [comment]
Hi, may I have a look at your specific setting regarding the results in your last post?

On Oct 31, 2018, at 04:03, Tianlin-Gao <notifications@github.com<mailto:notifications@github.com>> wrote:


BTW, I use https://github.com/corenel/pytorch-adda<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_corenel_pytorch-2Dadda&d=DwMCaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=KUScIrWv9DcqHOllgaD0uQ&m=7ghePg-YJ1SoumiNtku-yZNINB1H7o1Kvt-YtJIVNQg&s=r-yT-qiAalhw6wb-kWIG2mO2cRLO0bT411figYpfwaQ&e=> not this offical Tensorflow. My result (A->W) is 81% using above setting.

**Comment by LamForest** [comment]
> Hi Tianlin-Gao,
> 
> Thanks for your reply. May I know which layer are you matching and do you fix some layers in the source/target encoder to avoid overfitting? Also, can you tell me the number/size of the hidden layer of your discriminator and which optimizer are you using, parameters for the optimizer? Or would you mind sharing your code for adda on Office?
> 
> I am sorry that this has been killing me for days. I just want to get adda work and build my model based on it. I would be really thankful if you can provide more information on that implementation.
> 
> Thanks,
> Rui
> [rw161@duke.edu](mailto:rw161@duke.edu)

Sorry for the delay. 
I use ResNet50 for source encoder and target encoder. For both of them, layers prior to Res4a are freezed. **Target encode is initialized from source encoder**. The output of avgpooling is used as the input for discriminator. The discriminator has 2 hidden layer. Each of them has 500 units.
The pytorch code for discriminator is as follow:
```
input_dims = 2048
hidden_dims = 500
output_dims = 2
         self.layer = nn.Sequential( 
             nn.Linear(input_dims, hidden_dims),
             nn.ReLU(),
             nn.Linear(hidden_dims, hidden_dims),
             nn.ReLU(),
             nn.Linear(hidden_dims, output_dims),
             nn.LogSoftmax()
         )
```
**Discriminator is from scratch. Source encoder is initialized from ImageNet ResNet50**
